### PR TITLE
Fix environment crash

### PR DESCRIPTION
**Pull request by pablog**
## Summary ##

Fix: https://github.com/Wolox/ios-base-project/issues/8

- Setting environment using compiler flags instead of using environment variables, since they are not injected when the application is not executed from xcode.

- Adding `isUITestTarget` in `Environment`.

### CODE REVIEW

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 0.0
:white_check_mark: Code Quality (Swiftlint): 100.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/ios-base-project/branches/fix-environment-crash) for more details.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 0.0
:white_check_mark: Code Quality (Swiftlint): 100.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/ios-base-project/branches/fix-environment-crash) for more details.

File: BaseProject/Environment.swift [hunk 1]
```diff
@@ -8,39 +8,59 @@
...
-        } else if environment == "test" {
-            return .test
-        } else {
-            fatalError("Environment variable ENVIRONMENT isn't set.")
-        }
+        #endif
+        #if RELEASE
+            return .release
+        #endif
+        fatalError("Swift compiler flag isn't set.")
```
Comment (danielaRiesgo) [inline BaseProject/Environment.swift]: `Swift environment compiler flag isn't set.`

File: BaseProject/Environment.swift [hunk 2]
```diff
@@ -8,39 +8,59 @@
...
     }
+
+}
+
+internal extension Environment {
+    
+    private static let EnvironmentKey = "ENVIRONMENT"
+    
+    // This property is set in scheme Test action's arguments (environment variables).
```
Comment (danielaRiesgo) [inline BaseProject/Environment.swift]: `This property should be set`

File: BaseProject/Environment.swift [hunk 3]
```diff
@@ -8,39 +8,59 @@
...
+    private static let EnvironmentTest = "Test"
+    
+    // This property is set in UITests by launchEnvironment arguments.
```
Comment (danielaRiesgo) [inline BaseProject/Environment.swift]: `This property should be set`

File: BaseProject/Environment.swift (hunk 1 above)
Comment (pablog) [inline BaseProject/Environment.swift]: Done

File: BaseProject/Environment.swift (hunk 2 above)
Comment (pablog) [inline BaseProject/Environment.swift]: Done

File: BaseProject/Environment.swift (hunk 3 above)
Comment (pablog) [inline BaseProject/Environment.swift]: Done

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 0.0
:white_check_mark: Code Quality (Swiftlint): 100.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/ios-base-project/branches/fix-environment-crash) for more details.

**Comment by pablog** [comment]
@danielaRiesgo In swift only `#if` works.

Take a look at [this](https://stackoverflow.com/questions/24003291/ifdef-replacement-in-the-swift-language/41632355#41632355)

Also, I moved the compiler flags to a section added in xcode 8 as that link suggests.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 0.0
:white_check_mark: Code Quality (Swiftlint): 100.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/ios-base-project/branches/fix-environment-crash) for more details.

**Pull request merged**